backend/danswer/connectors/alation/connector.py

```diff
@@ -0,0 +1,228 @@
+        """Refreshes the access token if it expires while we're making requests
+        TODO - Probably needs to also check the validate API to handle expiring
+        refresh tokens.
+        """
+        if (
+            self.token_expiry is None
+            or datetime.now(self.token_expiry.tzinfo) >= self.token_expiry
+        ):
+            self.authenticate()
+
+    def get_batch_for_object_type(
+        self, otype: AlationObjectType, limit: int = 100, skip: int = 0
+    ) -> list[dict[str, Any]]:
+        """A simplified function for making batch requests to get Alation objects.
+        Not all Alation APIs work this way, but this'll do for current objects.
+        """
+        self._validate_access_token()
+        response = self.api_client.get(
+            self._create_url_for_object(
+                otype, params={"limit": str(limit), "skip": str(skip)}
+            ),
+        )
+        response.raise_for_status()
+        return response.json()
+
+    @staticmethod
+    def get_title_from_page(page: dict, otype: AlationObjectType) -> str:
+        """A helper to get the title for an Alation object"""
+        if otype == AlationObjectType.ARTICLE:
+            if len(page["title"]) == 0:
+                return f"Untitled Article [{page['id']}]"
+            return page["title"]
+        else:
+            if len(page["title"]) == 0:
+                return page["key"]
+            return page["title"]
+
+    @staticmethod
+    def get_body_from_page(page: dict, otype: AlationObjectType) -> str:
+        """A helper to get some kind of document body from an Alation object"""
+        if otype == AlationObjectType.ARTICLE:
+            return page["body"]
+        else:
+            return page["description"]
+
+
+class AlationConnector(LoadConnector):
+    """Connector to the Alation Data Catalog
+
+    Arguments:
+    server_url -- URL to the Alation Server
+    object_types_to_index -- List of Alation Object Types to index
+    batch_size -- The maximum number of objects to index in 1 request
+    max_objects_to_index_per_type -- The maximum number of objects to index for a given
+    object type. This is to reduce the liklihood of rate-limiting
+    """
+
+    def __init__(
+        self,
+        server_url: str,
+        object_types_to_index: list[AlationObjectType] = ALL_ALATION_OBJECT_TYPES,
+        batch_size: int = INDEX_BATCH_SIZE,
+        max_objects_to_index_per_type: int = 0,
+    ) -> None:
+        self.batch_size = batch_size
+        self.server_url = server_url
+        self.object_types_to_index = object_types_to_index
+        self.max_objects_to_index_per_type = max_objects_to_index_per_type
+
+    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
+        user_id = credentials["alation_user_id"]
+        refresh_token = credentials["alation_refresh_token"]
+        self.client = AlationClient(self.server_url, user_id, refresh_token)
```

Comment:
Declare this one in __init__ as None so the check later doesn't fail with self not having attr client